components/Cart/interfaces.rst

```diff
@@ -0,0 +1,102 @@
+~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by model representing a single CartItem.
+
+.. node::
+    This interface extends the :ref:`component_order_model_order-item-interface` |br|
+    For more detailed information go to `Sylius API's CartItemInterface`_.
+
+.. _Sylius API's CartItemInterface: http://api.sylius.org/Sylius/Component/Cart/Model/CartItemInterface.html
+
+Service Interfaces
+------------------
+
+.. _component_cart_provider_cart-provider-interface:
+
+CartProviderInterface
+~~~~~~~~~~~~~~~~~~~~~
+
+A cart provider retrieves existing cart or create new one based on the storage. To characterize an object which is a **Provider**,
+it needs to implement the ``CartProviderInterface``.
+
+.. note::
+    For more detailed information go to `Sylius Api CartProviderInterface`_.
+
+.. _Sylius Api CartProviderInterface: http://api.sylius.org/Sylius/Component/Cart/Provider/CartProviderInterface.html
+
+.. _component_cart_purger_purger-interface:
+
+PurgerInterface
+~~~~~~~~~~~~~~~
+
+A cart purger purges all expired carts. To characterize an object which is a **Purger**, it needs to implement the ``PurgerInterface``.
+
+.. note::
+    For more detailed information go to `Sylius Api PurgerInterface`_.
+
+.. _Sylius Api PurgerInterface: http://api.sylius.org/Sylius/Component/Cart/Purger/PurgerInterface.html
+
+.. _component_cart_resolver_item-resolver-interface:
+
+ItemResolverInterface
+~~~~~~~~~~~~~~~~~~~~~
+
+A cart resolver returns cart item that needs to be added based on given data. To characterize an object which is a **Resolver**,
+it needs to implement the ``ItemResolverInterface``.
+
+.. note::
+    For more detailed information go to `Sylius Api ItemResolverInterface`_.
+
+.. _Sylius Api ItemResolverInterface: http://api.sylius.org/Sylius/Component/Cart/Resolver/ItemResolverInterface.html
+
+.. caution::
+    This method throws `ItemResolvingException`_ if an error occurs.
+
+.. _ItemResolvingException: http://api.sylius.org/Sylius/Component/Cart/Resolver/ItemResolvingException.html
+
+.. _component_cart_repository_cart-repository-interface:
+
+CartRepositoryInterface
+~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to decouple from storage that provides expired carts, you should create repository class which implements this interface.
+
+.. note::
+    This interface extends the :ref:`component_order_repository_order-repository-interface` |br|
+    For more detailed information go to `Sylius API's CartRepositoryInterface`_.
+
+.. _Sylius API's CartRepositoryInterface: http://api.sylius.org/Sylius/Component/Cart/Repository/CartRepositoryInterface.html
+
+CartContextInterface
+~~~~~~~~~~~~~~~~~~~~
+
+This interface is implemented by the services responsible for setting and retrieving current cart identifier based on storage.
+To characterize an object which is a **CartContext** it needs to implement the ``CartContextInterface``
+
+.. note::
+    For more detailed information go to `Sylius API CartContextInterface`_.
```

Comment:
You use many variant of text "sylius api..". Could you choose one and use in all files?